cpp/daal/src/algorithms/dbscan/dbscan_dense_default_distr_impl.i

```diff
@@ -235,15 +245,22 @@ Status DBSCANDistrStep4Kernel<algorithmFPType, method, cpu>::compute(const DataC
 
     int result = 0;
 
-    size_t splitDim = -1;
+    size_t splitDim   = -1;
+    size_t valuedPart = 0;
     for (size_t part = 0; part < nBlocks; part++)
     {
         NumericTablePtr ntPartialSplit = NumericTable::cast((*dcPartialSplits)[part]);
         ReadRows<algorithmFPType, cpu> partialSplitRows(ntPartialSplit.get(), 0, 1);
         DAAL_CHECK_BLOCK_STATUS(partialSplitRows);
         const algorithmFPType * const partialSplit = partialSplitRows.get();
 
-        partialSplitValues[part] = partialSplit[0];
+        if ((size_t)partialSplit[1] == -1)
```

Comment:
Consider re-writing this if(). It has hard-to-understand comparison of unsigned size_t value to a signed const -1.  I'd rewrite to something like:
const size_t noSplitDim = -1;
...
if ((size_t)partialSplit[1] == noSplitDim)